be/src/exec/pipeline/sort/sort_context.cpp

```diff
@@ -37,7 +37,12 @@ ChunkPtr SortContext::pull_chunk() {
 
 Status SortContext::_merge_inputs() {
     int64_t total_rows = _total_rows.load(std::memory_order_relaxed);
-    int64_t require_rows = ((_limit < 0) ? total_rows : std::min(_limit, total_rows));
+
+    // Keep all the data if topn type is RANK or DENSE_RANK
+    int64_t require_rows = total_rows;
+    if (_topn_type == TTopNType::ROW_NUMBER) {
+        require_rows = ((_limit < 0) ? total_rows : std::min(_limit, total_rows));
```

Comment:
`_limit <=0` ?

Reply:
Ok, I'll update, but actually, _limit can only be -1 or positive value